core/src/main/scala/org/apache/spark/shuffle/ShuffleBlockPusher.scala

```diff
@@ -100,7 +106,15 @@ private[spark] class ShuffleBlockPusher(conf: SparkConf) extends Logging {
     pushRequests ++= Utils.randomize(requests)
 
     submitTask(() => {
-      pushUpToMax()
+      try {
+        pushUpToMax()
+      } catch {
+        case e: FileNotFoundException =>
```

Comment:
```
def handleResult(result: PushResult): Unit = {
        submitTask(() => {
          if (updateStateAndCheckIfPushMore(
            sizeMap(result.blockId), address, remainingBlocks, result)) {
            pushUpToMax()
          }
        })
```

We should also catch the exception there?

Reply:
Sure, I will catch it there as well because there is a chance that `pushUpToMax()` there can throw FNF. We mostly saw the `result` from push with FNF as failure so `updateStateAndCheckIfPushMore(...)` would return false. But it might just happen, that the files got deleted after that.